src/transactions/TransactionFrame.cpp

```diff
@@ -304,12 +305,29 @@ TransactionFrame::processSignatures(SignatureChecker& signatureChecker,
                           "transaction")
                 .Mark();
             markResultFailed();
-            return false;
+            allValid = false;
```

Comment:
We should not mark those metrics several times for a given transaction.
I think it's better to move the code contained here into the `if (!allValid)` block below (and only use this place to set `allValid`)